packages/cli/src/commands/config/set.ts

```diff
@@ -1,28 +1,79 @@
+import { failWith, printValueMap } from '../../utils/cli'
+import { CeloConfig, ConfigRetriever, defaultCeloConfig } from '../../utils/config'
 
 export default class Set extends LocalCommand {
   static description = 'Configure running node information for propogating transactions to network'
 
   static flags = {
-    ...LocalCommand.flags,
+    ...BaseCommand.flags,
     // Overrides base command node flag.
     node: flags.string({
       char: 'n',
-      required: true,
       description: 'URL of the node to run commands against',
       default: 'http://localhost:8545',
     }),
+    privateKey: flags.string({
+      description:
+        'Set to use a specific privateKey. Beware this privateKey will be stored on disk.',
+    }),
+    add: flags.boolean({
+      default: false,
+      description:
+        'If --add flag is set, it will add the configurations to the already saved cache. Otherwise will replace the old cache',
+    }),
   }
 
-  static examples = ['set  --node ws://localhost:2500', 'set  --node <geth-location>/geth.ipc']
+  static examples = [
+    'set  --node ws://localhost:2500',
+    'set  --node <geth-location>/geth.ipc',
+    'set  --node https://somehost.com  --useLedger',
+    'set  --useLedger  --ledgerCustomAddress "[0, 47, 99]"  --add',
+    'set  --node ws://localhost:2500  --useLedger  --ledgerAddresses 3 --ledgerConfirmAddress',
+    'set  --useAKV  --azureVaultName some-vault',
+  ]
 
   async run() {
     const res = this.parse(Set)
-    const config: CeloConfig = {
-      nodeUrl: res.flags.node,
+    const configRetriever: ConfigRetriever = new ConfigRetriever(this.config.configDir)
+    const celoConfig = res.flags.add ? configRetriever.getConfig() : { ...defaultCeloConfig }
+    res.raw
+      .map((v) => (v as any).flag)
+      .filter((v) => v !== 'add')
+      .forEach((v) => ((celoConfig as any)[v] = (res.flags as any)[v]))
+    const errors = validateConfig(celoConfig)
+    if (!errors) {
+      configRetriever.writeConfig(celoConfig)
+      console.log('Configuration saved!')
+      printValueMap(celoConfig)
+    } else {
+      failWith(`${errors} There was a configuration incompatibility. No changes were saved`)
     }
-    await writeConfig(this.config.configDir, config)
   }
 }
+
+function validateConfig(celoConfig: CeloConfig): string | undefined {
+  let counter = celoConfig.useLedger ? 1 : 0
+  counter = celoConfig.useAKV ? counter + 1 : counter
+  counter = celoConfig.privateKey ? counter + 1 : counter
+  if (counter > 1) {
+    return "Can't have two or more of these flags stored: '--useLedger', '--useAKV' or '--privateKey'."
+  }
+  if (
+    (celoConfig.ledgerAddresses ||
+      celoConfig.ledgerCustomAddresses ||
+      celoConfig.ledgerConfirmAddress) &&
+    !celoConfig.useLedger
+  ) {
+    console.log(
+      "You will need to use the '--useLedger' flag in your transactions, to use the Ledger flags that you set. You could also store the '--useLedger' flag to your configuration"
```

Comment:
> to use the Ledger flags that you set.

 Unclear what this means

Reply:
This is a friendly warning because the user is setting the `--ledgerAddresses 5` for example, but is not adding the `--useLedger`.
I didn't required to add that flag, because there could be a use case where the user doesn't want to use the ledger in every transaction, so it could just add which are the address in the ledger that it will use, but then for every `celocli` command add or not the `--useLedger`
Example:
```
> celocli config:set --node <myNode> --ledgerCustomAddresses "[3,6]"
> celocli transfer:gold --privateKey <privateKeyForAnotherAddress> --from <privateKeyAddress> --to <someAddress> --value 1000
> celocli transfer:gold --from <ledgerAddressIndex3> --to <someAddress> --value 1000 --useLedger
```

If the user set the `useLedger` in the first command, this will tell that the user wants to use the EVERY TIME, and the second command will fail.

Reply:
Writing this response I realise that I could add a very small change that could make everything easier, hahaha.
So, let me add something first...